src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1330,17 +1330,19 @@ public virtual ISession OpenSession()
 			{
 				_log.Debug("Opening Hibernate Session.");
 				var session = new SessionImpl(_sessionFactory, this);
-				if (_interceptor != null)
-				{
-					// NH specific feature
-					// _interceptor may be the shared accros threads EmptyInterceptor.Instance, but that is
-					// not an issue, SetSession is no-op on it.
-					_interceptor.SetSession(session);
-				}
+				SetSessionOnInterceptor(session);
 
 				return session;
 			}
 
+			protected virtual void SetSessionOnInterceptor(SessionImpl session)
```

Comment:
I think the correct approach is to drop this "feature". Currently the intention of `IInterceptor.SetSession` is not clear. I think it was an attempt to implement logging, which shall be done with a new "child" session instead. 

This "feature" was sneakily added without leaving any trace in the issue tracker.